contrib/pkg/clusterpool/clusterpool.go

```diff
@@ -0,0 +1,464 @@
+			return err
+		}
+	}
+	for _, obj := range objs {
+		accessor, err := meta.Accessor(obj)
+		if err != nil {
+			log.WithError(err).Errorf("Cannot create accessor for object of type %T", obj)
+			return err
+		}
+		accessor.SetNamespace(o.Namespace)
+		if _, err := rh.ApplyRuntimeObject(obj, scheme.Scheme); err != nil {
+			return err
+		}
+
+	}
+	return nil
+}
+
+func (o *ClusterPoolOptions) generateObjects() ([]runtime.Object, error) {
+	pullSecret, err := utils.GetPullSecret(o.PullSecret, o.PullSecretFile)
+	if err != nil {
+		return nil, err
+	}
+
+	if o.createCloudSecret {
+		switch o.Cloud {
+		case cloudAWS:
+			defaultCredsFilePath := filepath.Join(o.homeDir, ".aws", "credentials")
+			accessKeyID, secretAccessKey, err := awsutils.GetAWSCreds(o.CredsFile, defaultCredsFilePath)
+			if err != nil {
+				log.WithError(err).Error("Failed to get AWS credentials")
+				return nil, err
+			}
+			o.awsAccessKeyID = accessKeyID
+			o.awsSecretAccessKey = secretAccessKey
+		case cloudAzure:
+			creds, err := azureutils.GetCreds(o.CredsFile)
+			if err != nil {
+				log.WithError(err).Error("Failed to read in Azure credentials")
+				return nil, err
+			}
+			o.azureServicePrincipal = creds
+		case cloudGCP:
+			creds, err := gcputils.GetCreds(o.CredsFile)
+			if err != nil {
+				log.WithError(err).Error("Failed to get GCP credentials")
+				return nil, err
+			}
+			o.gcpServiceAccount = creds
+		}
+	}
+
+	imageSet, err := o.configureImages()
+	if err != nil {
+		return nil, err
+	}
+
+	result, err := o.build()
+	if err != nil {
+		return nil, err
+	}
+
+	if imageSet != nil {
+		result = append(result, imageSet)
+	}
+	if len(pullSecret) != 0 {
+		result = append(result, o.generatePullSecret(pullSecret))
+	}
+
+	return result, nil
+}
+
+func (o *ClusterPoolOptions) configureImages() (*hivev1.ClusterImageSet, error) {
+	if len(o.ClusterImageSet) > 0 {
+		return nil, nil
+	}
+
+	if o.ReleaseImage == "" {
+		if o.ReleaseImageSource == "" {
+			return nil, fmt.Errorf("specify either a release image or a release image source")
+		}
+		var err error
+		o.ReleaseImage, err = utils.DetermineReleaseImageFromSource(o.ReleaseImageSource)
+		if err != nil {
+			return nil, fmt.Errorf("cannot determine release image: %v", err)
+		}
+	}
+	imageSet := &hivev1.ClusterImageSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s-imageset", o.Name),
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ClusterImageSet",
+			APIVersion: hivev1.SchemeGroupVersion.String(),
+		},
+		Spec: hivev1.ClusterImageSetSpec{
+			ReleaseImage: o.ReleaseImage,
+		},
+	}
+	o.ClusterImageSet = imageSet.Name
+	return imageSet, nil
+}
+
+func (o *ClusterPoolOptions) build() ([]runtime.Object, error) {
+	// validate
+	if len(o.Name) == 0 {
+		return nil, fmt.Errorf("name is required")
+	}
+	if len(o.BaseDomain) == 0 {
+		return nil, fmt.Errorf("BaseDomain is required")
+	}
+	if len(o.ClusterImageSet) == 0 && len(o.ReleaseImage) == 0 {
```

Comment:
This makes it impossible for the user to use the `--release-image` flag since `o.ClusterImageSet` will be set in the `configureImages` function.